### PR TITLE
Random interface and roll implementation

### PR DESCRIPTION
**Pull request by Pawleshhh**
Add IRandomNumberGenerator with its default implementation that uses System.Random class.

Add Roll interfaces for rolling the dice. Two base implementations added:
- SingleRollBase - rolls one dice (can specify how many times), applies specified operator to given factor (e.g. D6 "- 3") and when division operator is used then a rounding operator can be specified (Round, Ceil, Floor).
- MultipleRollBase - contains different rolls and sums their outcome together.

### CODE REVIEW

File: Source/ProjectRPG.Core/Roll/Data/FloatRoundOperation.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+﻿namespace ProjectRPG.Core;
+
+public enum FloatRoundOperation
+{
+    Round,
+    Floor,
+    Ceil,
```
Comment (Mark01190) [inline Source/ProjectRPG.Core/Roll/Data/FloatRoundOperation.cs:7]: ,

File: Source/ProjectRPG.Core/Roll/SingleRollBase.cs [hunk 2]
```diff
@@ -0,0 +1,101 @@
+﻿namespace ProjectRPG.Core;
+
+internal class SingleRollBase : ISingleRoll
```
Comment (Mark01190) [inline Source/ProjectRPG.Core/Roll/SingleRollBase.cs:3]: why internal?

File: Source/ProjectRPG.Core/Random/DefaultRng.cs [hunk 3]
```diff
@@ -0,0 +1,48 @@
+﻿namespace ProjectRPG.Core;
+
+internal class DefaultRng : IRandomNumberGenerator
```
Comment (Mark01190) [inline Source/ProjectRPG.Core/Random/DefaultRng.cs:3]: again internal class?

File: Source/ProjectRPG.Core/Roll/SingleRollBase.cs (hunk 2 above)
Comment (Pawleshhh) [reply Source/ProjectRPG.Core/Roll/SingleRollBase.cs:3]: This is a base implementation that gets returned by static method in ISingleRoll interface. We don't want to show implementation, only the interface ;)

File: Source/ProjectRPG.Core/Random/DefaultRng.cs (hunk 3 above)
Comment (Pawleshhh) [reply Source/ProjectRPG.Core/Random/DefaultRng.cs:3]: We hide implementation under the interface. There will be special factory classes that will create specific rng classes in the future.

**Review by Mark01190 — APPROVED** [review]

**Pull request merged**